doc_src/history.txt

```diff
@@ -36,12 +37,18 @@ These flags can appear before or immediately after one of the sub-commands liste
 
 - `-e` or `--exact` searches or deletes items in the history that exactly match the specified text string. This is the default for the `--delete` flag. Note that the match is case-insensitive by default. If you really want an exact match, including letter case, you must use the `-C` or `--case-sensitive` flag.
 
+- `--file` changes the current history file. That option can be used both independent of any other option and in conjunction with the other options. In both cases the change to the current history file is persistent and affects subsequent commands. The switch to a new history file occurs before any other option is acted on and an implicit `history --save` occurs before the switch.
```

Comment:
This does not seem to work. `history --file banana` will just ignore the "--file" and search for "banana" instead.

Additionally, I'd prefer it if there was just one way to change the history file, and the variable seems to be nicer since it can also be set from outside (env FISH_HISTFILE=none fish).